sdk/communication/communication-administration/src/communicationIdentity/communicationIdentityClient.ts

```diff
@@ -70,7 +82,7 @@ export class CommunicationIdentityClient {
    */
   public constructor(
     connectionStringOrUrl: string,
-    credentialOrOptions?: KeyCredential | CommunicationIdentityOptions,
+    credentialOrOptions?: KeyCredential | CommunicationIdentityOptions | TokenCredential,
     maybeOptions: CommunicationIdentityOptions = {}
   ) {
     const { url, credential } = parseClientArguments(connectionStringOrUrl, credentialOrOptions);
```

Comment:
I wonder if `parseClientArguments` still handles this correctly after expanding with TokenCredential

Reply:
No it does not. Found this issue when implementing live tests. cc @0rland0Wats0n 
Do you have a suggestion better having `parseClientArguments` return 
```
export type UrlWithCredential = {
  url: string;
  credential?: KeyCredential;
  tokenCredential?: TokenCredential;
};
```
Or using `unknown` type and casting it downstream later?

Reply:
We found out it doesn't during working on the live tests with @0rland0Wats0n 
we made `parseClientArguments` credential be `KeyCredential | TokenCredential` but had to change a few things down stream in SMS and Phone Number to case credential as KeyCredential. 
Thoughts?